### PR TITLE
Adding Basic User Interface to OOP School Library

### PR DESCRIPTION
**Pull request by ticoniq**
### Overview:
This pull request introduces a basic user interface to our Object-Oriented Programming (OOP) school library application, enhancing the user experience and enabling seamless interactions with the library system.

### Changes Made:
- Implemented a simple command-line interface (CLI) for users to interact with the library.
- Created menus and prompts for actions such as browsing books, borrowing, and returning books.
- Enhanced the `Library` class to accommodate user requests through the interface.
- Designed clear user prompts and informative messages to guide interactions.

### CODE REVIEW

**Review by joel-kalema — COMMENTED** [review]
Hi @ticoniq 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request  🛑 

pirates-of-the-caribbean-johnny-depp


- [ ] You have submitted a project that has linters that does not pass, make sure you solve them and the ask again for a review


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @joel-kalema in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: classes/main.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  main(app)
+end
+
+def create_rental(app)
+  app.create_rental
+  main(app)
+end
+
+puts 'Welcome to School Library App!'
+main(app)
```
Comment (AbayomiOlaoye) [inline classes/main.rb:84]: - [ ] Looks like your app is expecting some arguments when it's being run, and I'd like to think this is not part of the requirement. Upon running `ruby main.rb`, it should open the entire interface and options available in the app. Check the error message below have a grasp:
image

- You might as well consider using a `Switch/Case` conditional rendering for your app, something like this:
```
# main.rb
def main
  app = App.new()

  response = nil

  # [OPTIONAL] you can remind the student about the ruby lessons about basic sintax like loops: https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/lessons/basic_syntax.md  
  # *Present the user with a list of options to perform.
  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    # *Lets users choose an option.
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      # *Have a way to quit the app.
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()
```

File: classes/app.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+require './teacher'
+require './classroom'
+require './rental'
+require './book'
+
+class App
+  def initialize()
+    @all_books = []
+    @all_persons = []
+  end
```
Comment (AbayomiOlaoye) [inline classes/app.rb:12]: - [ ] Great job, @ticoniq but I believe that the requirement also asked that you create `rentals` too just as shown in the requirement below and that makes me think that a variable called `rentals` is best created here, not in other file:
image

**Review by AbayomiOlaoye — CHANGES_REQUESTED** [review]
## STATUS: CHANGE REQUIRED ♻️ 

Hi @ticoniq :wave: , 

Good job with this milestone so far! What a brilliant attempt!!
There are just a few more issues to work on before moving on to the next project but you are almost there!

so_close

## Highlights
- Descriptive Pull Request message :heavy_check_mark: 
- Good use of GitHub flow :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- No linter error :heavy_check_mark:
- Fairly clean logic :heavy_check_mark: 
- Descriptive ReadMe documentation :heavy_check_mark: 

## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- You could also consider writing descriptive commit messages to convey more meaning and substance about the changes made in each commit.:
image


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @ticoniq, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Rubocop Linters setup properly✔️
- School Library App properly setup ✔️
- App works really great ✔️
- Add the basic UI for the School library app ✔️
- Good use of Git flow ✔️

## Optional changes

- [ ] Kindly check your commit messages and always be sure to have them written in a professional way. Here is a [link](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8) for more info

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**